kernel/parser.h

```diff
@@ -3,17 +3,30 @@
 
 #include <stddef.h>
 
+/// This container stores a string split up by whitespace
+/// and quotation canaries in order to tokenize the
+/// internal string by allowing it to be treated
+/// as multiple smaller strings by virtue of the
+/// indices provided by the struct.
+///
+/// The `str` member contains the internal string, and
+/// the `indices` member points to an array of `char *`
+/// which point to each token within the internal `str`.
+/// The size attribute describes the size of the array
+/// pointer to by `indices`.
 struct tokenized_string {
     char* str;
     char** indices;
     size_t size;
 };
 
-/*
+/// frees a `tokenized_string` struct and its subsequent member pointers
 void free_tokenized_string(struct tokenized_string *tsp);
-void free_tokenized_string(struct tokenized_string tsv);
-*/
 
+/// Instantiates a `tokenized_string` with the string passed to it.
```

Comment:
same thing here, i think the explanation could just be shortened down to a single line
also, no need for punctuation here, but it's up to you

Reply:
I made the following changes to this branch. It should be good to push to the main branch if you have no other suggestions.